mitiq/shadows/shadows_utils.py

```diff
@@ -8,13 +8,74 @@
 # LICENSE file in the root directory of this source tree.
 
 """Defines utility functions for classical shadows protocol."""
-from typing import List, Tuple
+from typing import Tuple, List, Any
 
 import numpy as np
 from numpy.typing import NDArray
+import cirq
 
 import mitiq
 
+from itertools import product
+
+Paulis = [
+    cirq.I._unitary_(),
+    cirq.X._unitary_(),
+    cirq.Y._unitary_(),
+    cirq.Z._unitary_(),
+]
+
+
+def kronecker_product(matrices: List[NDArray[Any]]) -> NDArray[Any]:
+    """
+    Returns the Kronecker product of a list of matrices.
+    """
+    result = matrices[0]
+    for matrix in matrices[1:]:
+        result = np.kron(result, matrix)
+    return result
+
+
+# Input must be a 2x2 matrix, output is a vector of 4 elements
+def operator_ptm_vector_rep(opt: NDArray[Any]) -> NDArray[Any]:
+    r"""
+    Returns the PTM vector representation
+    :math:`|o\rangle\!\rangle\in \mathcal{H}_{4^n}`
+    of an operator :math:`opt\in \mathcal{L}(\mathcal{H}_{2^n})`.
+    """
+    # vector i-th entry is math:`d^{-1/2}Tr(oP_i)`
+    # where P_i is the i-th Pauli matrix
+    assert (
+        len(opt.shape) == 2 and opt.shape[0] == opt.shape[1]
+    ), "Input must be a square matrix"
+    num_qubits = int(np.log2(opt.shape[0]))
+    opt_vec = []
+    for pauli_combination in product(Paulis, repeat=num_qubits):
+        kron_product = kronecker_product(pauli_combination)
+        opt_vec.append(
+            np.trace(opt @ kron_product) * np.sqrt(1 / 2**num_qubits)
+        )
+    return np.array(opt_vec)
+
+
+def eigenvalues_to_bitstring(values: List[int]) -> str:
+    """Converts eigenvalues to bitstring. e.g., [-1,1,1] -> '100'"""
+    return "".join(["1" if v == -1 else "0" for v in values])
+
+
+def bitstring_to_eigenvalues(bitstring: str) -> List[int]:
+    """Converts bitstring to eigenvalues. e.g., '100' -> [-1,1,1]"""
+    return [1 if b == "0" else -1 for b in bitstring]
+
+
+def create_string(n: int, loc_list: List[int]) -> str:
```

Comment:
Let's avoid single letter argument variable names. Maybe `string_length`? In a similar vein, what does `loc_list` mean? Location list? Can we make this more explicit?